moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -81,7 +81,9 @@ bool RuckigSmoothing::applySmoothing(robot_trajectory::RobotTrajectory& trajecto
     return false;
   }
 
-  return runRuckig(trajectory, ruckig_input);
+  auto ruckig_result = runRuckigInBatches(num_waypoints, trajectory, ruckig_input);
+  trajectory = ruckig_result.value();
+  return ruckig_result.has_value();
```

Comment:
Will this segfault if no value is returned?
https://en.cppreference.com/w/cpp/utility/optional/value

Wouldn't this work better?
```suggestion
  if(ruckig_result.has_value()){
    trajectory = ruckig_result.value();
  }
  return false; // Ruckig failed to smooth the trajectory
```